order/models.py

```diff
@@ -49,13 +50,22 @@ class Declaration(models.Model):
     is_paid = models.BooleanField(default=False)
     penalty_status = models.BooleanField(default=False)
     penalty = models.DecimalField(max_digits=4, decimal_places=2, default=0, null=True, blank=True)
-
     
-    def save(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__status = self.status.id if self.status else None
+        return 
+    
+    def save(self, force_insert=False, force_update=False, *args, **kwargs):
         
         azn_rate = Currency.objects.filter(name="AZN").values_list('rate', flat=True).first()
         usd_rate = Currency.objects.filter(name="USD").values_list('rate', flat=True).first()
 
+        if self.id and self.status:
+            if self.status.id != self.__status:
+                new_status_history = StatusHistory.objects.create(declaration = self, old_status = Status.objects.filter(id=self.__status).first(), new_status = Status.objects.filter(id=self.status.id).first())
```

Comment:
Instead of doing two db qqueries, you can do:
`new_status_history = StatusHistory.objects.create(declaration=self, old_status_id=self.__status, new_status_id=self.status.id)` which will still create FK relation